tests/auxiliary/minicore.rs

```diff
@@ -0,0 +1,116 @@
+//! Auxiliary `minicore` prelude which stubs out std preludes for tests that need to work in
+//! cross-compilation scenarios where no `std` is available (that don't need to `-Zbuild-std`).
+
+#![feature(no_core, lang_items, rustc_attrs, unsized_fn_params, transparent_unions)]
+#![allow(unused, improper_ctypes_definitions, internal_features)]
+#![no_std]
+#![no_core]
+
+#[lang = "sized"]
+pub trait Sized {}
+
+#[lang = "receiver"]
+pub trait Receiver {}
+impl<T: ?Sized> Receiver for &T {}
+impl<T: ?Sized> Receiver for &mut T {}
+
+#[lang = "copy"]
+pub trait Copy: Sized {}
+impl Copy for i32 {}
+impl Copy for f32 {}
+impl<T: ?Sized> Copy for &T {}
+impl<T: ?Sized> Copy for *const T {}
+impl<T: ?Sized> Copy for *mut T {}
+
+#[lang = "clone"]
+pub trait Clone: Sized {
+    fn clone(&self) -> Self;
+}
+
+#[lang = "phantom_data"]
+pub struct PhantomData<T: ?Sized>;
+impl<T: ?Sized> Copy for PhantomData<T> {}
+
+#[lang = "unsafe_cell"]
+#[repr(transparent)]
+pub struct UnsafeCell<T: ?Sized> {
+    value: T,
+}
+
+pub trait Any: 'static {}
+
+pub enum Option<T> {
+    None,
+    Some(T),
+}
+impl<T: Copy> Copy for Option<T> {}
+
+pub enum Result<T, E> {
+    Ok(T),
+    Err(E),
+}
+impl<T: Copy, E: Copy> Copy for Result<T, E> {}
+
+#[lang = "manually_drop"]
+#[repr(transparent)]
+pub struct ManuallyDrop<T: ?Sized> {
+    value: T,
+}
+impl<T: Copy + ?Sized> Copy for ManuallyDrop<T> {}
+
+#[repr(transparent)]
+#[rustc_layout_scalar_valid_range_start(1)]
+#[rustc_nonnull_optimization_guaranteed]
+pub struct NonNull<T: ?Sized> {
+    pointer: *const T,
+}
+impl<T: ?Sized> Copy for NonNull<T> {}
+
+#[repr(transparent)]
+#[rustc_layout_scalar_valid_range_start(1)]
+#[rustc_nonnull_optimization_guaranteed]
+pub struct NonZero<T>(T);
+
+// This just stands in for a non-trivial type.
+pub struct Vec<T> {
+    ptr: NonNull<T>,
+    cap: usize,
+    len: usize,
+}
+
+pub struct Unique<T: ?Sized> {
+    pub pointer: NonNull<T>,
+    pub _marker: PhantomData<T>,
+}
+
+#[lang = "global_alloc_ty"]
+pub struct Global;
+
+#[lang = "owned_box"]
+pub struct Box<T: ?Sized, A = Global>(Unique<T>, A);
+
+#[repr(C)]
+struct RcBox<T: ?Sized> {
+    strong: UnsafeCell<usize>,
+    weak: UnsafeCell<usize>,
+    value: T,
+}
+pub struct Rc<T: ?Sized, A = Global> {
```

Comment:
I think Vec, Rc and Arc should be kept out of minicore. They are defined in liballoc, your versions are not functional at all and compatibility.rs is probably the only test that uses them, while many other tests can take advantage of minicore.

Reply:
Good point, I'll remove them when I revisit this.

Reply:
Removed `alloc` items.